docker/Dockerfile

```diff
@@ -25,6 +26,7 @@ RUN cd musica \
     && cmake -S . \
              -B build \
              -D CMAKE_BUILD_TYPE=${BUILD_TYPE} \
+             -D MUSICA_GIT_TAG=${MUSICA_GIT_TAG} \
```

Comment:
This is primarily for the fetch content integration tests, because if the git tag isn't included in the cmake args, it defaults to main (even if you're working on a development branch). I added it to all the tests, just so the GH actions can treat all the Docker builds the same.

For most of the tests, you probably don't have to specify the git tag. For the ones that do need it (the fetch content ones), you can pass in the branch name when you build the image:
```
docker build -t musica . -f docker/Dockerfile.gcc-integration --build-arg MUSICA_GIT_TAG=my_dev_branch
```

Reply:
Okay, this is for the fetch content integration. I understand it better